idaes/core/util/testing.py

```diff
@@ -409,3 +414,36 @@ def get_reaction_rate_basis(b):
             return MaterialFlowBasis.mass
         else:
             return MaterialFlowBasis.other
+
+
+def _enable_scip_solver_for_testing(
+    name: str = "scip",
+) -> Union[Callable[[], None], None]:
+    ampl_module_scip, is_available = attempt_import("ampl_module_scip")
+    if not is_available:
+        _log.warning(
+            "ampl_module_scip must be installed to enable SCIP solver for testing"
+        )
+        return None
+
+    new_path_entry = ampl_module_scip.bin_dir
+    # TODO prepending new_path_entry to PATH means that the "testing" SCIP will "shadow"
+    # existing executable directories already on PATH
+    # this behavior would match the (implied) semantics of this function, i.e.
+    # "enable SCIP solver used for testing (at the expense of other non-testing SCIP executables)"
+    os.environ["PATH"] = os.pathsep.join([new_path_entry, os.environ["PATH"]])
+    Executable.rehash()
+
+    def remove_from_path():
+        path_as_list = os.environ["PATH"].split(os.pathsep)
+
+        try:
+            path_as_list.remove(new_path_entry)
+        except ValueError:
```

Comment:
It's not a bad idea to add a test to cover this case to future-proof against any changes in how AMPL and/or Python handle their `PATH` entries / modifications in the future.

Reply:
I agree it'd be nice to have the "undo" functionality tested. It's a bit tricky but I'll see what I can do

Reply:
@mrmundt I've added a basic test for the PATH manipulation: https://github.com/IDAES/idaes-pse/pull/1282/commits/986ba84bbc6b51640fa3a590083dde10523f2685